internal/provider/utils/utils.go

```diff
@@ -22,12 +22,13 @@ func NamespacedName(obj client.Object) types.NamespacedName {
 	}
 }
 
-// GetHashedName returns a partially hashed name for the string including up to 48 characters of the original name before the hash
+// GetHashedName returns a partially hashed name for the string including up to 48 characters of the original name before the hash.
+// Input `nsName` should be formatted as `{Namespace}/{ResourceName}`.
 func GetHashedName(nsName string) string {
 
 	h := sha256.New() // Using sha256 instead of sha1 due to Blocklisted import crypto/sha1: weak cryptographic primitive (gosec)
-	hSum := h.Sum([]byte(nsName))
-	hashedName := strings.ToLower(fmt.Sprintf("%x", hSum))
+	h.Write([]byte(nsName))
```

Comment:
can you help elaborate why this is needed ?

Reply:
here is a [go playground example](https://go.dev/play/p/NaRpThfqDRT), may help you see why.

Reply:
cool thanks for highlighting this !